packages/amplify_core/lib/src/category/amplify_auth_category.dart

```diff
@@ -46,6 +48,68 @@ class AuthCategory extends AmplifyCategory<AuthPluginInterface> {
 
   /// {@template amplify_core.amplify_auth_category.sign_up}
   /// Create a new user with the given [username] and [password].
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="imports"?>
+  /// ```dart
+  /// import 'package:amplify_auth_cognito/amplify_auth_cognito.dart';
+  /// import 'package:amplify_flutter/amplify_flutter.dart';
+  /// ```
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="signup"?>
+  /// ```dart
+  /// Future<void> signUpUser({
+  ///   required String username,
+  ///   required String password,
+  ///   required String email,
+  ///   String? phoneNumber,
+  /// }) async {
+  ///   try {
+  ///     final userAttributes = {
+  ///       AuthUserAttributeKey.email: email,
+  ///       if (phoneNumber != null) AuthUserAttributeKey.phoneNumber: phoneNumber,
+  ///       // additional attributes as needed
+  ///     };
+  ///     final result = await Amplify.Auth.signUp(
+  ///       username: username,
+  ///       password: password,
+  ///       options: SignUpOptions(
+  ///         userAttributes: userAttributes,
+  ///       ),
+  ///     );
+  ///     return _handleSignUpResult(result);
+  ///   } on AuthException catch (e) {
+  ///     safePrint('Error signing up user: ${e.message}');
+  ///   }
+  /// }
+  /// ```
+  ///
+  /// The next step in the sign up flow is to confirm the user. A confirmation code
+  /// will be sent to the email or phone number provided during sign up. Prompt the
+  /// user to check their device for the code sent by Cognito.
+  ///
+  /// <?code-excerpt "doc/lib/auth.dart" region="handle-signup"?>
+  /// ```dart
+  /// Future<void> _handleSignUpResult(SignUpResult result) async {
+  ///   switch (result.nextStep.signUpStep) {
+  ///     case AuthSignUpStep.confirmSignUp:
+  ///       final codeDeliveryDetails = result.nextStep.codeDeliveryDetails!;
+  ///       _handleCodeDelivery(codeDeliveryDetails);
```

Comment:
Did you intend for `_handleCodeDelivery` to be included in the doc comment as well? I think it is fine to exclude, but wasn't sure if you had intended it to be included here.